### PR TITLE
Aravis several updates

### PR DESCRIPTION
**Pull request by ArkadiuszRaj**
Did small typo last time hence camera with 0-index is skipped.  My bad :(

As pushing correction with one character added is worthless ;) I  have created combined PR to address:

- accessing camera with index 0 (bug fix)
- adding control of Aravis transmission buffer size (write CAP_PROP_BUFFERSIZE)
- adding possibility to monitor current status of Aravis output buffer (read CAP_PROP_BUFFERSIZE)
- reading current image region size (CAP_PROP_FRAME_WIDTH & CAP_PROP_FRAME_HEIGHT)
- reading current frame time (in ms) sent by camera (CAP_PROP_POS_MSEC) as frameID/fps

Having frame ID directly from camera, I have also redesigned auto exposure to skip frames taken while exposure params have changed. Also if no change is needed but gain can be reduced, it is done as simply image quality is top prio. Now this code is much more stable.

I am in contact with Aravis author, thus in some time will push another update, to handle situation when camera gets accidentally disconnected.

### CODE REVIEW

File: modules/videoio/src/cap_aravis.cpp [hunk 1]
```diff
@@ -76,7 +82,7 @@
 #define MODE_Y800   CV_FOURCC_MACRO('Y','8','0','0')
 #define MODE_Y12    CV_FOURCC_MACRO('Y','1','2',' ')
 
-#define BETWEEN(a,b,c) ((a) < (b) ? (b) : ((a) > (c) ? (c) : (a) ))
+#define LIMIT(a,b,c) (cv::max(cv::min((a),(c)),(b)))
```
Comment (paroj) [inline modules/videoio/src/cap_aravis.cpp]: [lets call this clip](https://docs.scipy.org/doc/numpy/reference/generated/numpy.clip.html)

Comment (ArkadiuszRaj) [inline modules/videoio/src/cap_aravis.cpp]: Yeap, better name. Just changed that to optimize the code.

**Comment by alalek** [comment]
Please squash all commits into one when patch will be ready.

**Comment by ArkadiuszRaj** [comment]
Have some sort of problem with identifying the logic of VideoCapture. grabFrame() can only return false when communication problem is faced. Hence I assume that it is up to the opencv user to further deal with this, by for example closing high level app or trying to reconstruct VideoCapture object. 

That's said I am adding only small correction preventing calling Aravis functions after timeout happens.

**Comment by ArkadiuszRaj** [comment]
Patch finished. 

Tested on aravis-0.4 & aravis-0.6. 

Camera power disconnection is handled by returning NULL from grabFrame as stated above. 

Main application in loop tries to create VideoCapture object, which is destroyed when grabFrame fails. Works fine, no memory leaks observed.

**Comment by alalek** [comment]
:+1:

**Pull request merged**